src/main/java/org/codehaus/mojo/taglist/TagListReport.java

```diff
@@ -501,6 +523,30 @@ public List constructSourceDirs()
     }
 
     /**
+     * Get the files to include, as a comma separated list of patterns.
+     */
+    public String getIncludes()
+    {
+        if ( includes != null ) {
+            return String.join(",", includes);
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Get the files to exclude, as a comma separated list of patterns.
+     */
+    public String getExcludes()
+    {
+      if ( excludes != null ) {
+          return String.join(",", excludes);
+      } else {
+          return null;
+      }
+    }
+
```

Comment:
Those methods should match the parameter type (which is `String[]`), like so:

https://github.com/apache/maven-jar-plugin/blob/maven-jar-plugin-3.2.2/src/main/java/org/apache/maven/plugins/jar/AbstractJarMojo.java#L354-L370

This will also mean you'd need to move the join-logic over to the `FileAnalyzer.java` class or add another method.
But I think this perfectly makes sense.

Suggestion:
* Rename the methods so they do not match the bean getter specifiaction.
* make them not-public. Only this class and `FileAnalyzer` need access.

Reply:
Instead of returning null (which should be documented!) you could also return an empty String which would correspond to an empty List..

Reply:
@bmarwell I renamed the methods and return an empty string rather than `null`.